### ISSUE TITLE
Support Lets Encrypt wildcard certificate

### ISSUE THREAD

**Issue by alexAubin** [YunoHost/issues#2089]
We should definitely investigate what are the specificities of Lets Encrypt wildcard certs and be able to generate / use those automatically (eg when a wildcard A record exists) and use it for subdomains. That would heavily simplify creating subdomains.

**Comment by renne** [comment]
@alexAubin @tituspijean 
It's quite simple:
1. Fix the small bugs #2128 and #2134
2. Let's Encrypt wildcard certificates only work with the DNS challenge (usable since PR [1194](https://github.com/YunoHost/yunohost/pull/1194))
3. Use domain as subject distinguished name and sub- or wildcard domains as subject alternate names

**Comment by utkarshsethi** [comment]
have pushed [#1820](https://github.com/YunoHost/yunohost/pull/1820) for [#2128](https://github.com/YunoHost/issues/issues/2128)

**Comment by alexAubin** [comment]
>It's quite simple [...] wildcard certificates only work with the DNS challenge (usable since PR https://github.com/YunoHost/yunohost/pull/1194)

Then no it isn't "simple", or anything is simple on paper, and complex in pracice

The main technical challenge is integrating the DNS-based challenge in YunoHost like we did for the HTTP-based ACME challenge, which is required for wildcard certs, and changing the logistic to use certificates for subdomains etc

**Comment by jo-m1b** [comment]
Yes, that would be a major improvement for the security of YunoHost instances. A malicious hacker could easily scan the list of URLs from https://crt.sh/?q=ynh.fr&exclude=expired&group=none and test whether all those instances are up to date!

**Comment by samsam-rolon** [comment]
> Yes, that would be a major improvement for the security of YunoHost instances. A malicious hacker could easily scan the list of URLs from https://crt.sh/?q=ynh.fr&exclude=expired&group=none and test whether all those instances are up to date!

Yeah and with a challenge DNS, we were not exposing HTTP/80 port !
